src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.cs

```diff
@@ -2366,5 +2369,45 @@ void M([|D|]&lt;int&gt; lambda)
             var metadataAsSourceFile = await context.GenerateSourceAsync(navigationSymbol);
             context.VerifyResult(metadataAsSourceFile, expected);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+        public async Task TestFailedDecompilation()
+        {
+            var exportProvider = ExportProviderCache.GetOrCreateExportProviderFactory(TestExportProvider.EntireAssemblyCatalogWithCSharpAndVisualBasic.WithPart(typeof(ThrowingDecompiledSourceService))).CreateExportProvider();
+            var throwingService = exportProvider.GetExportedValue<ThrowingDecompiledSourceService>();
+            var expectedException = throwingService.Exception;
+            var lazyFailureText = new LazyToString(() => "// " + expectedException.ToString().Replace("\n", "\n// "));
+
+            var metadataSource = "public class C {}";
+            var symbolName = "C";
+
+            // Using the decompiler forces the language to C#, so the expected output is the same in both cases
+            IFormattable expected = $@"#region {FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
+// {CodeAnalysisResources.InMemoryAssembly}
+#endregion
+
+public class [|C|]
+{{
+    public C();
+}}
+{lazyFailureText}";
+
+            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, expected, allowDecompilation: true, exportProvider: exportProvider);
+            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.VisualBasic, expected, allowDecompilation: true, exportProvider: exportProvider);
```

Comment:
Consider making this a Theory that branches on language name.

Reply:
Current test matches all the other tests in this file. Wouldn't have objected to Theory but doesn't seem necessary to change it either.

Reply:
Ended up changing it to Theory